Mathlib/Combinatorics/SimpleGraph/Clique.lean

```diff
@@ -277,8 +334,88 @@ theorem cliqueFree_of_replaceVertex_cliqueFree [DecidableEq α] (s t : α) (h :
         simp_all
   · use ⟨f, hi⟩; simp_all
 
+@[simp]
+theorem cliqueFree_two : G.CliqueFree 2 ↔ G = ⊥ := by
+  classical
+  constructor
+  · simp_rw [← edgeSet_eq_empty, Set.eq_empty_iff_forall_not_mem, Sym2.forall, mem_edgeSet]
+    exact fun h a b hab => h _ ⟨by simpa [hab.ne], card_doubleton hab.ne⟩
+  · rintro rfl
+    exact cliqueFree_bot le_rfl
+#align simple_graph.clique_free_two SimpleGraph.cliqueFree_two
+
 end CliqueFree
 
+section CliqueFreeOn
+variable {s s₁ s₂ : Set α} {t : Finset α} {a b : α} {m n : ℕ}
+
+/-- `G.CliqueFreeOn s n` means that `G` has no `n`-cliques contained in `s`. -/
+def CliqueFreeOn (G : SimpleGraph α) (s : Set α) (n : ℕ) : Prop :=
+  ∀ ⦃t⦄, ↑t ⊆ s → ¬G.IsNClique n t
+#align simple_graph.clique_free_on SimpleGraph.CliqueFreeOn
+
+theorem CliqueFreeOn.subset (hs : s₁ ⊆ s₂) (h₂ : G.CliqueFreeOn s₂ n) : G.CliqueFreeOn s₁ n :=
+  fun _t hts => h₂ <| hts.trans hs
+#align simple_graph.clique_free_on.subset SimpleGraph.CliqueFreeOn.subset
+
+theorem CliqueFreeOn.mono (hmn : m ≤ n) (hG : G.CliqueFreeOn s m) : G.CliqueFreeOn s n := by
+  rintro t hts ht
+  obtain ⟨u, hut, hu⟩ := t.exists_smaller_set _ (hmn.trans ht.card_eq.ge)
+  exact hG ((coe_subset.2 hut).trans hts) ⟨ht.clique.subset hut, hu⟩
+#align simple_graph.clique_free_on.mono SimpleGraph.CliqueFreeOn.mono
+
+theorem CliqueFreeOn.anti (hGH : G ≤ H) (hH : H.CliqueFreeOn s n) : G.CliqueFreeOn s n :=
+  fun _t hts ht => hH hts <| ht.mono hGH
+#align simple_graph.clique_free_on.anti SimpleGraph.CliqueFreeOn.anti
+
+@[simp]
+theorem cliqueFreeOn_empty : G.CliqueFreeOn ∅ n ↔ n ≠ 0 := by
+  simp [CliqueFreeOn, Set.subset_empty_iff]
+#align simple_graph.clique_free_on_empty SimpleGraph.cliqueFreeOn_empty
+
+@[simp]
+theorem cliqueFreeOn_singleton : G.CliqueFreeOn {a} n ↔ 1 < n := by
+  obtain _ | _ | n := n <;>
+    simp [CliqueFreeOn, isNClique_iff, ←subset_singleton_iff', (Nat.succ_ne_zero _).symm]
+
+#align simple_graph.clique_free_on_singleton SimpleGraph.cliqueFreeOn_singleton
+
+@[simp]
+theorem cliqueFreeOn_univ : G.CliqueFreeOn Set.univ n ↔ G.CliqueFree n := by
+  simp [CliqueFree, CliqueFreeOn]
+#align simple_graph.clique_free_on_univ SimpleGraph.cliqueFreeOn_univ
+
+protected theorem CliqueFree.cliqueFreeOn (hG : G.CliqueFree n) : G.CliqueFreeOn s n :=
+  fun _t _ ↦ hG _
+#align simple_graph.clique_free.clique_free_on SimpleGraph.CliqueFree.cliqueFreeOn
+
+theorem cliqueFreeOn_of_card_lt {s : Finset α} (h : s.card < n) : G.CliqueFreeOn s n :=
+  fun _t hts ht => h.not_le <| ht.2.symm.trans_le <| card_mono hts
+#align simple_graph.clique_free_on_of_card_lt SimpleGraph.cliqueFreeOn_of_card_lt
+
+--TOOD: Restate using `simple_graph.indep_set` once we have it
```

Comment:
```suggestion
-- TODO: Restate using `SimpleGraph.IndepSet` once we have it
```
or is `IndepSet` a function not a type?

Reply:
It's currently nothing, but you're right nonetheless.